src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioDiagnosticAnalyzerExecutor.cs

```diff
@@ -151,6 +151,11 @@ private CompilationWithAnalyzers CreateAnalyzerDriver(CompilationWithAnalyzers a
 
         private void ReportAnalyzerExceptions(Project project, ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<DiagnosticData>> exceptions)
         {
+            if (exceptions == null)
```

Comment:
But the contract isn't obvious. Dealing with failure is something we should *force* the code to think about in an unambiguous fashion, at the point of the failure, i.e. where the call to the Invoke() originally happened. The fact you're changing the contract here and _not_ having to update all the call sites to match is terribly concerning to me.

Put another way, rather than a comment in a PR that says "I've checked all the call sites", I'd rather we changed the API to something that makes the failure clear, and I see all the call sites being updated to check the "it failed" flag. Then the code stands by itself in proving the assertion we updated stuff.